### ISSUE TITLE
Crash when debugging on local device after the splash screen disappear

### ISSUE THREAD

**Issue by sanmuru** [dotnet/maui#4824]
### Description

App crash when I debug it on my local Android device. The splash screen is fine but then the App crash.
I try many times and the crash happens constantly.

### Steps to Reproduce

1. Create a File > New .NET MAUI App
2. Make no changes
3. Connect up local Android device
4. Click 'Debug' button
5. Deploy process is OK
6. App startup and the splash screen appear
7. Splash screen fade out and at this moment the App crash

### Version with bug

Preview 12

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Visual Studio 17.2.0-pre.1.0, Google Android SDK Platform 31, Android 11.0 - API 30(local device)

### Did you find any workaround?

It seems that in debug mode the crash happens constantly, but everything is OK when I open the app (which was installed through deploy process automatically) on my Android device. The only workaround I could tell is do not debug the MAUI App with local device.

### Relevant log output

```shell
> am start -a "android.intent.action.MAIN" -c "android.intent.category.LAUNCHER" -n "com.companyname.mauiapp1/crc64e632a077a20c694c.MainActivity"
> Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=com.companyname.mauiapp1/crc64e632a077a20c694c.MainActivity }
[Zygote] Unable to open libbeluga.so: dlopen failed: library "libbeluga.so" not found.
[nyname.mauiapp] Late-enabling -Xcheck:jni
[nyname.mauiapp] Unquickening 21 vdex files!
[nyname.mauiapp] Unknown bits set in runtime_flags: 0x40000000
[RefClass] java.lang.reflect.InvocationTargetException
[OneTrace] Mark active for pid=21907? true
[oplus.android.OplusFrameworkFactoryImpl] get feature:IOplusScreenModeFeature
[oplus.android.OplusFrameworkFactoryImpl] getOplusScreenModeFeature
[Perf] Connecting to perf service.
[NetworkSecurityConfig] No Network Security Config specified, using platform default
[NetworkSecurityConfig] No Network Security Config specified, using platform default
[debug-app-helper] Checking if libmonodroid was unpacked to /data/app/~~19GKZeQ7U188EaFhR6jlEw==/com.companyname.mauiapp1-J-Chv_H6pu2tQTFDJKh3rA==/lib/arm64/libmonodroid.so
[debug-app-helper] Native libs extracted to /data/app/~~19GKZeQ7U188EaFhR6jlEw==/com.companyname.mauiapp1-J-Chv_H6pu2tQTFDJKh3rA==/lib/arm64, assuming application/android:extractNativeLibs == true
[debug-app-helper] Setting up for DSO lookup in app data directories
[debug-app-helper] Added filesystem DSO lookup location: /data/app/~~19GKZeQ7U188EaFhR6jlEw==/com.companyname.mauiapp1-J-Chv_H6pu2tQTFDJKh3rA==/lib/arm64
[debug-app-helper] Using runtime path: /data/app/~~19GKZeQ7U188EaFhR6jlEw==/com.companyname.mauiapp1-J-Chv_H6pu2tQTFDJKh3rA==/lib/arm64
[debug-app-helper] checking directory: `/data/user/0/com.companyname.mauiapp1/files/.__override__/lib`
[debug-app-helper] directory does not exist: `/data/user/0/com.companyname.mauiapp1/files/.__override__/lib`
[debug-app-helper] Checking whether Mono runtime exists at: /data/user/0/com.companyname.mauiapp1/files/.__override__/libmonosgen-2.0.so
[debug-app-helper] Checking whether Mono runtime exists at: /data/app/~~19GKZeQ7U188EaFhR6jlEw==/com.companyname.mauiapp1-J-Chv_H6pu2tQTFDJKh3rA==/lib/arm64/libmonosgen-2.0.so
[debug-app-helper] Mono runtime found at: /data/app/~~19GKZeQ7U188EaFhR6jlEw==/com.companyname.mauiapp1-J-Chv_H6pu2tQTFDJKh3rA==/lib/arm64/libmonosgen-2.0.so
[nyname.mauiapp] Attempt to remove non-JNI local reference, dumping thread
[DOTNET] JNI_OnLoad: JNI_OnLoad in pal_jni.c
[monodroid] Creating public update directory: `/data/user/0/com.companyname.mauiapp1/files/.__override__`
[nyname.mauiapp] Attempt to remove non-JNI local reference, dumping thread
[monodroid-debug] Trying to initialize the debugger with options: --debugger-agent=transport=dt_socket,loglevel=0,address=127.0.0.1:8867,server=y,embedding=1
[monodroid-assembly] open_from_bundles: failed to load assembly MauiApp1.dll
[monodroid-gc] GREF GC Threshold: 46080
[monodroid-assembly] open_from_bundles: failed to load assembly Mono.Android.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Runtime.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Java.Interop.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Collections.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/MauiApp1.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/Mono.Android.dll [External]
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/System.Runtime.dll [External]
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/Java.Interop.dll [External]
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/System.Collections.dll [External]
Resolved pending breakpoint for 'Android.Runtime.JNIEnv.RegisterJniNatives(System.IntPtr, System.Int32, System.IntPtr, System.IntPtr, System.Int32)' to /Users/builder/azdo/_work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNIEnv.cs:124 [0x00000].
[monodroid-assembly] open_from_bundles: failed to load assembly System.Threading.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/System.Threading.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Runtime.InteropServices.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/System.Runtime.InteropServices.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Threading.Thread.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Diagnostics.StackTrace.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/System.Threading.Thread.dll [External]
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/System.Diagnostics.StackTrace.dll [External]
[nyname.mauiapp] Attempt to remove non-JNI local reference, dumping thread
Loaded assembly: data-0xb400007487ae4ec0 [External]
Loaded assembly: data-0xb400007487afc4c0 [External]
Loaded assembly: data-0xb400007487b14c40 [External]
Loaded assembly: data-0xb400007487b8c500 [External]
[monodroid-assembly] open_from_bundles: failed to load assembly netstandard.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/netstandard.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Linq.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/System.Linq.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.VisualStudio.DesignTools.TapContract.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.VisualStudio.DesignTools.TapContract.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Xamarin.HotReload.Contracts.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Xamarin.HotReload.Contracts.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.Collections.Concurrent.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/System.Collections.Concurrent.dll [External]
Resolved pending breakpoint for 'Xamarin.HotReload.HotReloadAgent.BreakpointSendToIde(System.String)' to D:\a\_work\1\s\HotReload\Source\Xamarin.HotReload.Agent\HotReloadAgent.cs:415 [0x00000].
[monodroid-assembly] open_from_bundles: failed to load assembly System.Text.Json.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/System.Text.Json.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.Text.Encodings.Web.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/System.Text.Encodings.Web.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly System.ComponentModel.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/System.ComponentModel.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.Maui.Controls.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Xamarin.HotReload.Contracts.dll
Loaded assembly: /data/data/com.companyname.mauiapp1/files/.__override__/Microsoft.Maui.Controls.dll [External]
[monodroid-assembly] open_from_bundles: failed to load assembly Xamarin.HotReload.Contracts.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.VisualStudio.DesignTools.TapContract.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.VisualStudio.DesignTools.TapContract.dll
[monodroid-assembly] open_from_bundles: failed to load assembly System.ObjectModel.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.Maui.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.Maui.Graphics.dll
[monodroid-assembly] open_from_bundles: failed to load assembly Microsoft.Extensions.Logging.Abstractions.dll
[nyname.mauiapp] Could not load signature of Microsoft.Maui.Controls.HotReload.Forms.MauiAgent:VisualTreeChanged due to: Could not resolve type with token 01000079 from typeref (expected class 'Microsoft.Maui.VisualTreeChangeEventArgs' in assembly 'Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null') assembly:Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null type:Microsoft.Maui.VisualTreeChangeEventArgs member:(null)
[libc] Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x8 in tid 21907 (nyname.mauiapp1), pid 21907 (nyname.mauiapp1)
```

**Comment by drasticactions** [comment]
```
nyname.mauiapp] Could not load signature of Microsoft.Maui.Controls.HotReload.Forms.MauiAgent:VisualTreeChanged due to: Could not resolve type with token 01000079 from typeref (expected class 'Microsoft.Maui.VisualTreeChangeEventArgs' in assembly 'Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null') assembly:Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null type:Microsoft.Maui.VisualTreeChangeEventArgs member:(null)
```

You need to update to Preview 13. Your app is throwing that error because MAUI changed the location of VisualTreeChangeEventArgs from Maui.Controls to Core, and that had to be matched for XAML Hot Reload. Because you're running Preview 12, it has the old type, which is throwing.

Unless you're forcing it to version 12, it should have automatically updated to 13. I would check your dotnet workloads and see what version your running. Otherwise, I would say make sure you're on the newest version of VS and the newest MAUI preview.

**Comment by sanmuru** [comment]
That makes sense.
Yesterday I installed Visual Studio 17.2.0-pre.1.0 and then used [.NET MAUI Check tool v0.10.0.0](https://github.com/Redth/dotnet-maui-check/releases/tag/0.10.0) to check out if everything was ready, and it said MAUI Preview 12 was missing. I let the tool fix those missing components and probably overwrote Preview 13 that was already installed via VS 17.2.0-pre.1.0.
Now I've updated to Preview 13, and the debug startup crash never happen any more.
